app/helpers/application_helper.rb

```diff
@@ -1,4 +1,19 @@
 module ApplicationHelper
+
+  def root_path
+    if user_signed_in?
+      if is_rover?(current_user)
+        welcome_rovers_path
+      elsif is_admin?(current_user)
+        dashboard_path
```

Comment:
If an admin is also a rover, they will be redirected to welcome_rover_page. It would be better to check for admin first:
``` 
if is_admin?(current_user)
  dashboard_path
elsif is_rover?(current_user)
  welcome_rovers_path

```